### PR TITLE
coursier: prefer the primary_url over mirrors

### PR DESCRIPTION
**Pull request by dhalperi**
This is for https://github.com/bazelbuild/rules_jvm_external/issues/349 and tangentially for https://github.com/bazelbuild/rules_jvm_external/issues/723.

Users are often confused by (expected) 404s when using multiple Maven repositories. These 404s can be safely ignored (Bazel's repository fetching logic gracefully tries all mirrors), but they are chatty and there is as of yet no way to silence them.

See also: https://github.com/bazelbuild/bazel/issues/13394

While we cannot solve the issue, we can reduce its impact by ordering the URLs: `http_file` is documented to try them in order:
https://bazel.build/rules/lib/repo/http#http_file-urls

### CODE REVIEW

File: coursier.bzl [hunk 1]
```diff
@@ -1043,6 +1043,10 @@ def _coursier_fetch_impl(repository_ctx):
         primary_artifact_path = infer_artifact_path_from_primary_and_repos(primary_url, repository_urls)
 
         mirror_urls = [url + "/" + primary_artifact_path for url in repository_urls]
+        if primary_url in mirror_urls:
```
Comment (dhalperi) [inline coursier.bzl:1046]: It wasn't obvious to me that this is guaranteed to be the case, but I also didn't see how it couldn't be.

**Review by dhalperi — COMMENTED** [review]
FYI, I am assuming there will be tests to update, and that will help me figure out a new test to write as well.

File: coursier.bzl (hunk 1 above)
Comment (dhalperi) [reply coursier.bzl:1046]: Ahh, auth makes it not true.

File: tests/custom_maven_install/regression_testing_install.json [hunk 2]
```diff
@@ -2175,34 +2175,34 @@
                 "url": "https://repo1.maven.org/maven2/com/microsoft/azure/adal4j/1.6.0/adal4j-1.6.0.jar"
             },
             {
-                "coord": "com.nimbusds:lang-tag:1.6",
+                "coord": "com.nimbusds:lang-tag:1.7",
```
Comment (dhalperi) [inline tests/custom_maven_install/regression_testing_install.json:2178]: FYI repinning just did this upgrade to 1.7. I can add another version set to fix if preferred.

**Review by jin — APPROVED** [review]

**Comment by jin** [comment]
Nice!

**Pull request merged**